app/scripts/modules/amazon/instance/awsInstanceType.service.js

```diff
@@ -214,6 +214,59 @@ module.exports = angular.module('spinnaker.aws.instanceType.service', [
         });
     };
 
+    let instanceSizeHierarchy = ['xlarge', 'large', 'medium', 'small', 'micro', 'nano'];
+
+    function sortTypesByFamilyAndSize(o1, o2) {
+      var type1 = o1.split('.'),
+          type2 = o2.split('.');
+
+      if (type1.length != 2 || type2.length != 2) {
```

Comment:
Can you change `!=` to `!==` (and `==` to `===`) everywhere? Linter should be warning on this.